community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPagedFile.java

```diff
@@ -47,12 +51,12 @@
             UnsafeUtil.getFieldOffset( MuninnPagedFile.class, "lastPageId" );
 
     final MuninnPageCache pageCache;
-    // This is the table where we translate file-page-ids to cache-page-ids:
     final int pageSize;
     final PageCacheTracer tracer;
 
-    final PrimitiveLongObjectMap<MuninnPage>[] translationTables;
-    final StampedLock[] translationTableLocks;
+    // This is the table where we translate file-page-ids to cache-page-ids:
+    volatile Object[][] translationTable;
+    final StampedLock translationTableResizeLock;
```

Comment:
I'm mainly curious about why this particular lock type was chosen.
I would have reached for a monitor lock first, because of it's syntactical simplicity. The actual lock type doesn't matter much, but I'd like the choice to be well considered.